torch/_dynamo/compiled_autograd.py

```diff
@@ -147,9 +147,11 @@ def post_acc_grad_hook(self, inputs, hook_id):
             inputs,
         )
         with disable_proxy_modes_tracing():
-            inputs = [maybe_clone(x) for x in inputs]
+            if isinstance(inputs, list):
+                inputs = [maybe_clone(x) for x in inputs]
+            else:
+                inputs = [maybe_clone(inputs)]
```

Comment:
@jansel Note - I actually don't think `def pre_hook(self, inputs, hook_id):` can also have the same bug - we always make a list on the CA calling side.